sklearn/cluster/tests/test_optics.py

```diff
@@ -97,6 +98,9 @@ def test_extract_xi():
                             -1, [4] * 5]
     X, expected_labels = shuffle(X, expected_labels, random_state=rng)
 
+    if _IS_32BIT:
+        X = X.astype(np.float32)
```

Comment:
float64 should still work on 32bit arch. If it fails it may mean that something internally in OPTICS creates arrays of dtype `np.int` (or `np.intp`) instead of `X.dtype`...

Reply:
Time for a deep dive into OPTICS. I hope I have better vision when I surface.